tests/bluetooth/audio/ascs/src/test_ase_control_params.c

```diff
@@ -0,0 +1,953 @@
+					     sizeof(data_expected));
+}
+
+ZTEST_F(test_ase_control_params, test_config_qos_number_of_ases_0x00)
+{
+	uint8_t ase_id = test_ase_id_get(fixture->ase);
+	const uint8_t buf[] = {
+		0x02,                   /* Opcode = Config QoS */
+		0x00,                   /* Number_of_ASEs */
+		ase_id,                 /* ASE_ID[0] */
+		0x01,                   /* CIG_ID[0] */
+		0x01,                   /* CIS_ID[0] */
+		0xFF, 0x00, 0x00,       /* SDU_Interval[0] */
+		0x00,                   /* Framing[0] */
+		0x02,                   /* PHY[0] */
+		0x64, 0x00,             /* Max_SDU[0] */
+		0x02,                   /* Retransmission_Number[0] */
+		0x0A, 0x00,             /* Max_Transport_Latency[0] */
+		0x40, 0x9C, 0x00,       /* Presentation_Delay[0] */
+	};
+
+	test_config_qos_expect_invalid_length(&fixture->conn, ase_id, fixture->ase_cp,
+					      &fixture->stream, buf, sizeof(buf));
+}
+
+ZTEST_F(test_ase_control_params, test_config_qos_too_short)
+{
+	uint8_t ase_id = test_ase_id_get(fixture->ase);
+	const uint8_t buf[] = {
+		0x02,                   /* Opcode = Config QoS */
+		0x01,                   /* Number_of_ASEs */
+		ase_id,                 /* ASE_ID[0] */
+		0x01,                   /* CIG_ID[0] */
+		0x01,                   /* CIS_ID[0] */
+		0xFF, 0x00, 0x00,       /* SDU_Interval[0] */
+		0x00,                   /* Framing[0] */
+		0x02,                   /* PHY[0] */
+		0x64, 0x00,             /* Max_SDU[0] */
+		0x02,                   /* Retransmission_Number[0] */
+		0x0A, 0x00,             /* Max_Transport_Latency[0] */
+		0x40, 0x9C, 0x00,       /* Presentation_Delay[0] */
+		0x00,
+	};
+
+	test_config_qos_expect_invalid_length(&fixture->conn, ase_id, fixture->ase_cp,
+					      &fixture->stream, buf, sizeof(buf));
+}
+
+ZTEST_F(test_ase_control_params, test_config_qos_too_long)
+{
+	uint8_t ase_id = test_ase_id_get(fixture->ase);
+	const uint8_t buf[] = {
+		0x02,                   /* Opcode = Config QoS */
+		0x01,                   /* Number_of_ASEs */
+		ase_id,                 /* ASE_ID[0] */
+		0x01,                   /* CIG_ID[0] */
+		0x01,                   /* CIS_ID[0] */
+		0xFF, 0x00, 0x00,       /* SDU_Interval[0] */
+		0x00,                   /* Framing[0] */
+		0x02,                   /* PHY[0] */
+		0x64, 0x00,             /* Max_SDU[0] */
+		0x02,                   /* Retransmission_Number[0] */
+		0x0A, 0x00,             /* Max_Transport_Latency[0] */
+		0x40, 0x9C,             /* Presentation_Delay[0] */
+	};
+
+	test_config_qos_expect_invalid_length(&fixture->conn, ase_id, fixture->ase_cp,
+					      &fixture->stream, buf, sizeof(buf));
+}
+
+static void test_enable_expect_invalid_length(struct bt_conn *conn, uint8_t ase_id,
+					      const struct bt_gatt_attr *ase_cp,
+					      struct bt_bap_stream *stream,
+					      const uint8_t *buf, size_t len)
+{
+	const uint8_t data_expected[] = {
+		0x03,           /* Opcode */
```

Comment:
@fredrikdanebjer had a comment in one of your other PRs about including `ascs_internal` and using the opcodes from that. Also applies here